### PR TITLE
Literature tagtree: hide no-reference tags

### PR DESCRIPTION
**Pull request by rabstejnek**

### CODE REVIEW

File: frontend/summary/summary/LiteratureTagtree.js [hunk 1]
```diff
@@ -3,60 +3,115 @@ import BaseVisual from "./BaseVisual";
 import TagTree from "lit/TagTree";
 import TagTreeViz from "lit/TagTreeViz";
 import HAWCModal from "utils/HAWCModal";
+import ReactDOM from "react-dom";
```
Comment (shapiromatron) [inline frontend/summary/summary/LiteratureTagtree.js:6]: use `shared/components/CheckboxInput` instead?

File: frontend/lit/TagTree.js [hunk 2]
```diff
@@ -64,6 +64,24 @@ class TagTree extends Observee {
         tag.parent.children.splice(index, 1);
     }
 
+    prune_no_references() {
```
Comment (shapiromatron) [inline frontend/lit/TagTree.js:67]: add comment to explain what this does

File: frontend/summary/summary/LiteratureTagtree.js [hunk 3]
```diff
@@ -3,60 +3,115 @@ import BaseVisual from "./BaseVisual";
...
+        // prune nodes with no refs from root node
+        if (this.data.settings.hide_empty) tagtree.prune_no_references();
+
+        new TagTreeViz(tagtree, $plotDiv, title, url);
+    }
+
+    buildViz($plotDiv, data) {
+        this.buildPlot($plotDiv, data);
+        this.buildCheckbox($plotDiv, () => {
+            this.data.settings.hide_empty = !this.data.settings.hide_empty;
```
Comment (shapiromatron) [inline frontend/summary/summary/LiteratureTagtree.js:102]: change `hide_empty` to `hide_empty_tag_nodes`

**Comment by rabstejnek** [comment]
image
Checkbox on custom image

image
Checkbox on lit review viz

**Comment by rabstejnek** [comment]
### Checkbox below viz
image
Smaller window

image
Larger window

**Comment by shapiromatron** [comment]
@rabstejnek do my changes look ok?

**Pull request merged**